sdv/datasets/demo.py

```diff
@@ -0,0 +1,140 @@
+
+
+def _get_dataset_url(modality, dataset_name):
+    return os.path.join(BUCKET_URL, modality.upper(), dataset_name + '.zip')
+
+
+def _validate_args_download_demo(modality, dataset_name, output_folder_name):
+    possible_modalities = ['single_table', 'multi_table', 'sequential']
+    if modality not in possible_modalities:
+        raise InvalidArgumentError(f"'modality' must be in {possible_modalities}.")
+
+    if output_folder_name and os.path.exists(output_folder_name):
+        raise InvalidArgumentError(
+            f"Folder '{output_folder_name}' already exists. Please specify a different name "
+            "or use 'load_csvs' to load from an existing folder."
+        )
+
+    dataset_url = _get_dataset_url(modality, dataset_name)
+    try:
+        urllib.request.urlopen(dataset_url)
+    except urllib.error.HTTPError:
+        # If the dataset exists in the wrong modality, raise an error
+        other_modalities = set(possible_modalities) - {modality}
+        for other_modality in other_modalities:
+            dataset_url = _get_dataset_url(other_modality, dataset_name)
+            try:
+                urllib.request.urlopen(dataset_url)
+                raise InvalidArgumentError(
+                    f"Dataset name '{dataset_name}' is a '{other_modality}' dataset. "
+                    f"Use 'load_{other_modality}_demo' to load this dataset."
+                )
+            except urllib.error.HTTPError:
+                pass
+
+        # If the dataset doesn't exist at all, raise different error
+        raise InvalidArgumentError(
+            f"Invalid dataset name '{dataset_name}'. "
+            "Use 'list_available_demos' to get a list of demo datasets."
+        )
+
+
+def _download(modality, dataset_name, output_folder_name):
+    dataset_url = _get_dataset_url(modality, dataset_name)
+
+    LOGGER.info(f'Downloading dataset {dataset_name} from {dataset_url}')
+    response = urllib.request.urlopen(dataset_url)
+    bytes_io = io.BytesIO(response.read())
+
+    LOGGER.info(f'Extracting dataset into {output_folder_name}')
+    with ZipFile(bytes_io) as zf:
+        os.makedirs(output_folder_name, exist_ok=True)
+        zf.extractall(output_folder_name)
+        os.remove(os.path.join(output_folder_name, 'metadata_v0.json'))
+        os.rename(
+            os.path.join(output_folder_name, 'metadata_v1.json'),
+            os.path.join(output_folder_name, 'metadata.json')
+        )
+
+
+def _get_data(modality, output_folder_name):
+    data = {}
+    for filename in os.listdir(output_folder_name):
+        if filename != 'metadata.json':
+            data_path = os.path.join(output_folder_name, filename)
+            data[filename] = pd.read_csv(data_path)
+
+    if modality != 'multi_table':
+        data = data.popitem()[1]
+
+    return data
+
+
+def _get_metadata(modality, output_folder_name):
+    metadata_path = os.path.join(output_folder_name, 'metadata.json')
```

Comment:
the metadatas are all titled `metadata_v1.json`

Reply:
I renamed them to `metadata` because it looked a little silly to have a `metadata_v1` if we don't have a `metadata_v0`. If you think it could cause some issue/confusion though, I'll just change it.

Reply:
Is this for downloading or uploading? I may have been confused. I just mean when downloading the metadata, the filename is `metadata_v1` and if you look for just `metadata` it won't exist

Reply:
Further down the code I rename it:

```python3
os.rename(
            os.path.join(output_folder_name, 'metadata_v1.json'),
            os.path.join(output_folder_name, 'metadata.json')
        )
```